### PR TITLE
Minor fixes.

### PR DESCRIPTION
**Pull request by jacqueswww**
### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by fubuloubu** [comment]
Why is LGTM doing a JS analysis?

**Comment by jacqueswww** [comment]
@fubuloubu https://github.com/vyperlang/vyper/blob/master/examples/wallet/sign.js

**Comment by fubuloubu** [comment]
Oh yeah. Can't we use `web3.SignableMessage` to get rid of that JS?

File: vyper/codegen/abi.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def abi_decode(lll_node, src, pos=None):
             # descend into the child tuple
             lll_ret.append(abi_decode(o, child_loc, pos=pos))
         else:
-            lll_ret.append(make_setter(o, src_loc))
+            lll_ret.append(make_setter(o, src_loc, location=src.location, pos=pos))
```
Comment (charles-cooper) [inline vyper/codegen/abi.py]: I think `make_setter`'s location refers to the destination location. I assumed it would always be `'memory'`.

**Comment by charles-cooper** [comment]
I'm not sure about making the `make_setter` location to the source location is correct. But I noticed there's an inconsistency in my code - sometimes I set location to 'memory' and sometimes to `o.location`. https://github.com/vyperlang/vyper/blob/1174f57065109c2c3e1c9fbf951ceb6f410e684c/vyper/codegen/abi.py#L376-L381. I think it should just always be `dst.location`.

**Comment by jacqueswww** [comment]
@charles-cooper please check now; and then we can merge?

File: vyper/codegen/abi.py [hunk 2]
```diff
@@ -373,12 +373,12 @@ def abi_encode(dst, lll_node, pos=None, bufsz=None, returns=False):
...
         elif isinstance(o.typ, BaseType):
             d = LLLnode(dst_loc, typ=o.typ, location='memory')
-            lll_ret.append(make_setter(d, o, "memory", pos))
+            lll_ret.append(make_setter(d, o, location=d.location, pos=pos))
         elif isinstance(o.typ, ByteArrayLike):
             d = LLLnode.from_list(dst_loc, typ=o.typ, location='memory')
             lll_ret.append(
                     ['seq',
-                        make_setter(d, o, o.location, pos=pos),
+                        make_setter(d, o, location=o.location,  pos=pos),
```
Comment (charles-cooper) [inline vyper/codegen/abi.py]: I think I originally imagined that location would always be memory but I guess we should just fix this to always be `dst.location`, and also set `d`'s location to `dst.location`.

**Pull request merged**